runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -81,6 +81,8 @@ void deactivateDevice()
 
 extern "C" {
 
+void __catalyst__host__rt__unrecoverable_error() { RT_FAIL("Unrecoverable error."); }
```

Comment:
If this function is currently called only within the context of async qnodes, will you interested in updating the message to better convey the association with the issue? 

Maybe :thinking: 
```suggestion
void __catalyst__host__rt__unrecoverable_error() { RT_FAIL("Unrecoverable error from asynchronous execution of multiple quantum programs."); }
```